### PR TITLE
ocm: added new placement rule builder and unit tests

### PR DESCRIPTION
**Pull request by klaskosk**
Added `NewPlacementRuleBuilder`  and a unit test for it.

### CODE REVIEW

**Review by kononovn — CHANGES_REQUESTED** [review]
Looks good just few minor comments to fix.

**Review by kononovn — APPROVED** [review]
/lgtm

**Review by sebrandon1 — COMMENTED** [review]
One comment about returns.

File: pkg/ocm/placementrule.go [hunk 1]
```diff
@@ -26,10 +26,47 @@ type PlacementRuleBuilder struct {
...
+		glog.V(100).Info("The name of the PlacementRule is empty")
+
+		builder.errorMsg = "placementrule's 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Info("The namespace of the PlacementRule is empty")
+
+		builder.errorMsg = "placementrule's 'nsname' cannot be empty"
+	}
```
Comment (sebrandon1) [inline pkg/ocm/placementrule.go:55]:
```suggestion
	if name == "" {
		glog.V(100).Info("The name of the PlacementRule is empty")

		builder.errorMsg = "placementrule's 'name' cannot be empty"
		
		return builder
	}

	if nsname == "" {
		glog.V(100).Info("The namespace of the PlacementRule is empty")

		builder.errorMsg = "placementrule's 'nsname' cannot be empty"
		
		return builder
	}
```

**Comment by klaskosk** [comment]
@sebrandon1 Thanks, I made the change to returns

File: pkg/ocm/placementrule.go (hunk 1 above)
Comment (sebrandon1) [reply pkg/ocm/placementrule.go]: @klaskosk After speaking with @kononovn we decided your original approach was fine for now.  If you want to revert my above suggestion that would be great.

**Review by kononovn — APPROVED** [review]

**Pull request merged**